lib/matplotlib/projections/polar.py

```diff
@@ -1447,12 +1447,23 @@ def format_sig(value, delta, opt, fmt):
                     cbook._g_sig_digits(value, delta))
             return f"{value:-{opt}.{prec}{fmt}}"
 
-        return ('\N{GREEK SMALL LETTER THETA}={}\N{GREEK SMALL LETTER PI} '
-                '({}\N{DEGREE SIGN}), r={}').format(
-                    format_sig(theta_halfturns, delta_t_halfturns, "", "f"),
-                    format_sig(theta_degrees, delta_t_degrees, "", "f"),
-                    format_sig(r, delta_r, "#", "g"),
-                )
+        # In case fmt_xdata was not specified, resort to default
+        if self.fmt_xdata is None:
+            return ('\N{GREEK SMALL LETTER THETA}={}\N{GREEK SMALL LETTER PI} '
+                    '({}\N{DEGREE SIGN}), r={}').format(
+                        format_sig(theta_halfturns, delta_t_halfturns, "", "f"),
+                        format_sig(theta_degrees, delta_t_degrees, "", "f"),
+                        format_sig(r, delta_r, "#", "g")
+                        if self.fmt_ydata is None
+                        else self.format_ydata(r)
```

Comment:
I was a bit confused by this portion as well when first glancing at this because it is duplicated below. I think this might be nice to bring outside the `fmt_xdata` if statement:

`r_label = ...` and then reference `r_label` here and the following code block.